Assets/Project/Modules/Enemies/Slime/Scripts/SlimeRefactor/EnemyPatrolling.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Unity.VisualScripting;
+using UnityEngine;
+
+namespace Popeye.Modules.Enemies.Components
+{
+    public class EnemyPatrolling : ANavMeshMovement
+    {
+        private PatrolType _patrolType = PatrolType.None;
+        private Transform[] _wayPoints;
+        private int _wayPointIndex;
+        [SerializeField] private float _playerDistanceThreshold;
+        [SerializeField] private float _wayPointDistanceThreshold;
+        private float _squaredWayPointDistanceThreshold;
+        private float _squaredPlayerDistanceThreshold;
+        private IEnemyMediator _mediator;
+        private Vector3 _target;
+        private bool _patrolling;
+        
+        public enum PatrolType
+        {
+            None,
+            FixedWaypoints,
+            Random
+        }
+        
+        public void Configure(IEnemyMediator slimeMediator)
+        {
+            _mediator = slimeMediator;
+        }
+        
+        public void Start()
+        {
+            _squaredPlayerDistanceThreshold = _playerDistanceThreshold * _playerDistanceThreshold;
+
+            if (_patrolType == PatrolType.None)
+            {
+                return;
+            }
+            else if (_patrolType == PatrolType.FixedWaypoints)
+            {
+                
+                
+            }
+            else if (_patrolType == PatrolType.Random)
+            {
+                
+            }
```

Comment:
What will this be used for?
You can add comments inside the ifs for short descriptoin of the intended uses

Reply:
It can be delete, I simply forgot to do so 💅